### ISSUE TITLE
Narrow numeric range from min/max/between rules (e.g. integer|min:1 → positive-int)

### ISSUE THREAD

**Issue by lptn** [psalm/laravel-psalm-plugin#1234]
## Summary

Extend the validation-rule type inference from #468 to:

1. **Narrow numeric range** from `min` / `max` / `between` / `size` / `gt` / `gte` / `lt` / `lte` rules, e.g. `integer|min:1` → `positive-int` (`int<1, max>`), `integer|between:0,24` → `int<0, 24>`.
2. **Cover the typed accessors** (`integer()`, `float()`, `boolean()`) in `ValidatedFieldReadResolver`, so a rule-narrowed type is returned instead of the plain Laravel stub type.

Both are needed to fix the reproducer below; today the field falls back to plain `int`.

## Reproducer

```php
final class AddItemToCartRequest extends FormRequest
{
    public function rules(): array
    {
        return [
            'portions' => ['bail', 'required', 'integer', 'min:1', /* ... */],
        ];
    }

    /** @return positive-int */
    public function getPortions(): int
    {
        return $this->integer('portions');
    }
}
```

Psalm reports:

```
ERROR: MoreSpecificReturnType
The declared return type 'int<1, max>' for ...::getPortions is more specific
than the inferred return type 'int' (see https://psalm.dev/070)
    /** @return positive-int */
```

The declared `positive-int` is correct (the validator guarantees `>= 1`), but the plugin infers plain `int`, so the accurate docblock is flagged as redundant/too-specific.

## Root cause (two gaps)

**1. `integer()` is not a narrowed accessor.** `ValidatedFieldReadResolver` only narrows these (`src/Handlers/Validation/ValidatedFieldReadResolver.php`):

```php
public const KEYED_ACCESSOR_METHODS = ['validated', 'input', 'string', 'str', 'array', 'collect'];
private const ACCESSOR_METHODS      = ['validated', 'validate', 'safe', 'input', 'string', 'str', 'array', 'collect'];
```

`integer`, `float`, `boolean` are absent, so `$this->integer('portions')` returns Laravel's stub `int` regardless of `rules()`.

**2. `ruleToType()` ignores numeric-range parameters.** In `src/Handlers/Validation/ValidationRuleAnalyzer.php`:

```php
'integer' => new Union([new TInt(), new TNumericString()]),
```

The `$param` argument is already threaded in (used by `in`/`enum`/`decimal`), but `min`/`max`/`between`/`size`/`gt`/`gte`/`lt`/`lte` do not contribute a range, so `integer|min:1` stays `int` rather than becoming `int<1, max>`.

## Proposed behaviour

| Rules | Inferred type |
|---|---|
| `integer` + `min:1` | `int<1, max>` (aka `positive-int`) |
| `integer` + `min:0` | `int<0, max>` |
| `integer` + `max:10` | `int<min, 10>` |
| `integer` + `between:0,24` | `int<0, 24>` |
| `integer` + `size:5` | `int<5, 5>` (i.e. `5`) |
| `integer` + `gt:0` | `int<1, max>` |
| `integer` + `gte:0` | `int<0, max>` |
| `integer` + `lt:100` | `int<min, 99>` |
| `integer` + `lte:100` | `int<min, 100>` |

Notes:
- For a bare `integer` (no bound rule) keep the current `int|numeric-string`.
- `min`/`max`/`size` are overloaded (string length, array count, file size). Only apply range narrowing when a numeric rule (`integer`/`numeric`) is present on the same field; otherwise leave `min`/`max` alone (they already work as string-length/array-count constraints, which the plugin does not model today anyway).
- `gt`/`gte`/`lt`/`lte` can take a field name rather than a literal (e.g. `gt:other_field`); only narrow when the parameter is an integer literal, otherwise fall back.

## Scope / caveat

Same "unvalidated field" caveat noted when `input()` narrowing was discussed in #468: typed accessors read arbitrary request input, not only validated keys. Narrowing `integer('portions')` from `rules()` assumes the accessed key is validated. This matches the existing decision to narrow `input()`, so extending it to `integer()`/`float()`/`boolean()` is consistent; flagging it explicitly in case the maintainer wants to scope the range narrowing to `validated('field')` only.

## Environment

- `psalm/plugin-laravel`: `dev-master` (`6ac7789`)
- `vimeo/psalm`: `7.0.0-beta19`
- Laravel 13, PHP 8.5

Related: #468 (completed rule→type inference), #624 (dot-notation nested shapes).
